arangod/RocksDBEngine/RocksDBBuilderIndex.cpp

```diff
@@ -152,14 +155,38 @@ Result fillIndexSingleThreaded(
     }
     numDocsWritten++;
 
-    if (numDocsWritten % 1024 == 0) {  // commit buffered writes
+    if (numDocsWritten > 0 &&
+        numDocsWritten % 1024 == 0) {  // commit buffered writes
+
       res = partiallyCommitInsertions(batch, rootDB, trxColl, docsProcessed,
                                       ridx, foreground);
       // cppcheck-suppress identicalConditionAfterEarlyExit
       if (res.fail()) {
         break;
       }
 
+      if (count > 0) {
+        double p =
+            docsProcessed.load(std::memory_order_relaxed) * 100.0 / count;
+        ridx.progress(p);
+        if (progress != nullptr) {
+          (*progress)(p);
+        }
+      }
+
+#ifdef ARANGODB_ENABLE_FAILURE_TESTS
+      TRI_IF_FAILURE("fillIndex::pause") {
+        std::cout << "fillIndex::pause" << std::endl;
```

Comment:
The print statements here should really be removed, even though they are only included inside a failure point.